scripts/benchmark_db/bench.rb

```diff
@@ -176,169 +169,164 @@ def snapshot_height(snapshot)
-
-    metrics = {}
-    args = build_artefacts(dry_run)
-
-    import_command = splice_args(@import_command, args)
-    metrics[:import] = exec_command(import_command, dry_run)
-
-    # Save db size just after import
-    metrics[:import][:db_size] = db_size unless dry_run
-
-    validate_command = splice_args(@validate_command, args)
-    metrics[:validate] = exec_command(validate_command, dry_run)
-
-    clean_db(dry_run)
-
-    @metrics = metrics
-  end
-
-  def target
-    './target/release/forest'
-  end
-
-  def build_command
-    ['cargo', 'build', '--release']
-  end
-
-  def initialize(name:, config: {})
-    @name = name
-    @config = config
-    @import_command = [
-      target, '--config', '%<c>s', '--encrypt-keystore', 'false', '--import-snapshot', '%<s>s', '--halt-after-import'
-    ]
-    @validate_command = [
-      target, '--config', '%<c>s', '--encrypt-keystore', 'false',
-      '--import-snapshot', '%<s>s', '--halt-after-import', '--skip-load', '--height', '%<h>s'
-    ]
-    @metrics = {}
-  end
+# Determine correct snapshot download URL path based on value of chain.
+def get_url(chain, url)
+  value = chain == 'mainnet' ? 'mainnet' : 'calibrationnet'
+  output = syscall('aria2c', '--dry-run', "https://snapshots.#{value}.filops.net/minimal/latest.zst")
+  url || output.match(%r{Redirecting to (https://.+?\d+_.+)})[1]
 end
 
-# Benchmark class for ParityDb
-class ParityDbBenchmark < Benchmark
-  def build_command
-    ['cargo', 'build', '--release', '--no-default-features', '--features', 'forest_fil_cns,paritydb']
-  end
+# Helper function to create assignments for `download_and_move` function.
+def download_and_move_assignments(url)
+  filename = url.match(/(\d+_.+)/)[1]
+  checksum_url = url.sub(/\.car\.zst/, '.sha256sum')
+  checksum_filename = checksum_url.match(/(\d+_.+)/)[1]
+  decompressed_filename = filename.sub(/\.car\.zst/, '.car')
+  [filename, checksum_url, checksum_filename, decompressed_filename]
 end
 
-# Benchmark class for ParityDb with Jemalloc
-class JemallocBenchmark < Benchmark
-  def build_command
-    ['cargo', 'build', '--release', '--no-default-features', '--features', 'forest_fil_cns,paritydb,jemalloc']
+# Create snapshot directory; download checksum and compressed snapshot;
+# decompress and verify snapshot; clean up helper files and return path to snapshot.
+def download_and_move(url, output_dir)
+  filename, checksum_url, checksum_filename, decompressed_filename = download_and_move_assignments(url)
+
+  # Must move to `WORKING_DIR` and create temporary folder for temp snapshot
+  # files, as filesystem `tmp` partition may not be large enough for this task.
+  Dir.chdir(WORKING_DIR)
+  snapshot_dir = Dir.pwd
+  FileUtils.mkdir_p('snapshot_dl_files')
+  Dir.chdir('snapshot_dl_files') do
+    # Download, decompress and verify checksums.
+    @logger.info 'Downloading checksum...'
+    syscall('aria2c', checksum_url)
+    @logger.info 'Downloading snapshot...'
+    syscall('aria2c', '-x5', url)
```

Comment:
please use longopt names in scripts - I don't want to have to check the man page to find out what `-x5` means